server/routes/charts.js

```diff
@@ -205,16 +206,32 @@ router
   .get(ensureAuthenticated, async (req, res) => {
     try {
       const { dataDb } = req.app.locals
+      const { chart_id } = req.params
+      const { userAccess } = req.session
       const chart = await dataDb
         .collection(collections.charts)
         .findOne(
-          { _id: new ObjectId(req.params.chart_id), owner: req.user },
+          { _id: new ObjectId(chart_id), owner: req.user },
           { projection: { _id: 0 } }
         )
 
       if (!chart) return res.redirect(routes.charts)
 
-      const { userAccess } = req.session
+      if (req.headers['content-type'] === 'text/csv') {
```

Comment:
should this diff be in the api route? IMO it isn't an api call and could be a regular GET request.

Reply:
Good catch. 
https://github.com/AMP-SCZ/dpdash/pull/287/commits/53c75c19215f81ea1f646fb5e98102019a21521f